usersession/agent/rest_api.go

```diff
@@ -46,3 +59,129 @@ func sessionInfo(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(m)
 }
+
+type serviceInstruction struct {
+	Action   string   `json:"action"`
+	Services []string `json:"services"`
+}
+
+var (
+	stopTimeout = time.Duration(timeout.DefaultTimeout)
+	killWait    = 5 * time.Second
+)
+
+func stopOneService(service string, sysd systemd.Systemd) error {
+	err := sysd.Stop(service, stopTimeout)
+	if err != nil && systemd.IsTimeout(err) {
+		// ignore errors for kill; nothing we'd do differently at this point
+		sysd.Kill(service, "TERM", "")
+		time.Sleep(killWait)
```

Comment:
Should we wait unconditionally? If the wait time is a bit larger then this will just wait even though the service may have reacted faster. I also wonder if we really need this. Systemd models behaviour for stopping services, along with any timeouts and cleanup necessary.

What we might do instead is kill stopping fails.

Reply:
I adapted this code from `stopService` in `wrappers/services.go`.  Looking at the documentation in systemd.kill(5), it isn't obvious to me that this is needed either.

Reply:
We're already forgoing support for the oldest systems here: Ubuntu 14.04 and CentOS/RHEL 7 don't instantiate a user mode systemd, so this code would never run on those releases.  I wonder if that raises the bar high enough that we can just rely on `stop` being reliable?